File: tok/line_randomizer.py
import random
import os
import fnmatch

i = 0
line_of_poem = []
corpus = []
corpus_content = {}
corpus_poetry_by_line = {}

# import file names for all txt files in the folder
for file in os.listdir('./'):
    if fnmatch.fnmatch(file, '*.txt'):
        corpus.append(file)

# read in the text from each text file as a list of lines, with new line characters stripped
for poem in corpus:
    with open(poem, 'r') as f:
        temp_text = f.readlines()
        corpus_content[poem] = []
        for line in temp_text:
            corpus_content[poem].append(line.rstrip('\n'))

# create a dictionary in which each key is a line number (starting with zero, because Python
# and each value is a list of lines, one from each imported poem
while i < 77:
    single_line_content_all_poems = []
    for poem in corpus:
        single_line_content_all_poems.append(corpus_content[poem][i])
    corpus_poetry_by_line[i] = single_line_content_all_poems
    i += 1

# assemble a new poem that takes one random version of each line
assembled_poem = []
i = 0
while i < 77:
    assembled_poem.append(random.choice(corpus_poetry_by_line[i]))
    i += 1

# print the newly created poem
for line in assembled_poem:
    line_out = ''
    for char in line:
        if char == '|':
            line_out += '\n'
        else:
            line_out += char
    print(line_out)

File: tok/line_randomizer.php
<?php

$i = 0;
$line_of_poem = array();
$corpus = array();
$corpus_content = array();
$corpus_poetry_by_line = array();

# import file names for all txt files in the folder
foreach (scandir('./') as $key => $file_name) {
  if (strpos($file_name, '.txt') !== false) {
    $corpus[] = $file_name;
  }
}

# read in the text from each text file as a list of lines
# with new line characters stripped
foreach($corpus as $key => $poem) {
  $poem_text = @fopen('./' . $poem, 'r');
  $corpus_content[$poem] = file('./' . $poem);
  }


# create a dictionary in which each key is a line number (starting with zero, because Python
# and each value is a list of lines, one from each imported poem
while ($i < 77) {
  $single_line_content_all_poems = array();
  foreach($corpus as $key => $poem) {
    $single_line_content_all_poems[] = $corpus_content[$poem][$i];
  }
  $corpus_poetry_by_line[$i] = $single_line_content_all_poems;
  $i++;
}

# assemble a new poem that takes one random version of each line
$assembled_poem = array();
foreach($corpus_poetry_by_line as $key => $line) {
  $random_key = array_rand(array_values($line), 1);
  $assembled_poem[] = $line[$random_key];
  $i++;
}

# print the newly created poem
foreach($assembled_poem as $line) {
  $chars = str_split($line);
  $line_out = '';
  foreach($chars as $index => $char) {
    if ($char == '|') {
      $line_out .= '\n';
    } else {
      $line_out .= $char;
    }
  }
  print($line_out);
}

/*
foreach($assembled_poem as $index => $line) {
  echo $line;
}
*/
?>
